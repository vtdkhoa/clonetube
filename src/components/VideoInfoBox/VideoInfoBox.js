import React, { Component, Fragment } from 'react'
import { Image, Button, Divider } from 'semantic-ui-react'
import './VideoInfoBox.scss'

class VideoInfoBox extends Component {
  constructor(props) {
    super(props)
    this.state = {
      collapsed: true
    }
  }

  onToggleCollapse = () => {
    this.setState(prevState => {
      return {
        collapsed: !prevState.collapsed
      }
    })
  }

  render() {
    let descriptionTextClass = 'collapsed'
    let buttonTitle = 'Show More'
    let descriptionParagraphs = (
      <div>
        <p>Para 1</p>
        <p>Para 2</p>
        <p>Para 3</p>
        <p>Para 4</p>
        <p>Para 5</p>
      </div>
    )

    if (!this.state.collapsed) {
      descriptionTextClass = 'expanded'
      buttonTitle = 'Show Less'
    }

    return (
      <Fragment>
        <div className="video-info-box">
          <Image
            className="channel-image"
            src="http://via.placeholder.com/48x48"
            circular
          />
          <div className="video-info">
            <div className="channel-name">Channel Name</div>
            <div className="video-publication-date">Sat, Jul 18 2020</div>
          </div>
          <Button color="youtube">10K Subscribe</Button>
          <div className="video-description">
            <div className={descriptionTextClass}>
              {descriptionParagraphs}
            </div>
            <Button
              compact
              onClick={this.onToggleCollapse}
            >{buttonTitle}</Button>
          </div>
        </div>
        <Divider/>
      </Fragment>
    )
  }
}

export default VideoInfoBox