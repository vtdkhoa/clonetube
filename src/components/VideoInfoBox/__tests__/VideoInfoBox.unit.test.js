import React from 'react'
import VideoInfoBox from '../VideoInfoBox'
import { shallow } from 'enzyme'

describe('VideoInfoBox', () => {
  test('renders collapsed', () => {
    const wrapper = shallow(<VideoInfoBox/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders expanded', () => {
    const wrapper = shallow(<VideoInfoBox/>)
    wrapper.setState({ collapsed: false })
    expect(wrapper).toMatchSnapshot()
  })
})