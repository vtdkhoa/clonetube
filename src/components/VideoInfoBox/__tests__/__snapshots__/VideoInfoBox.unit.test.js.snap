// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoInfoBox renders collapsed 1`] = `
<Fragment>
  <div
    className="video-info-box"
  >
    <Image
      as="img"
      circular={true}
      className="channel-image"
      src="http://via.placeholder.com/48x48"
      ui={true}
    />
    <div
      className="video-info"
    >
      <div
        className="channel-name"
      >
        Channel Name
      </div>
      <div
        className="video-publication-date"
      >
        Sat, Jul 18 2020
      </div>
    </div>
    <Button
      as="button"
      color="youtube"
    >
      10K Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="collapsed"
      >
        <div>
          <p>
            Para 1
          </p>
          <p>
            Para 2
          </p>
          <p>
            Para 3
          </p>
          <p>
            Para 4
          </p>
          <p>
            Para 5
          </p>
        </div>
      </div>
      <Button
        as="button"
        compact={true}
        onClick={[Function]}
      >
        Show More
      </Button>
    </div>
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoInfoBox renders expanded 1`] = `
<Fragment>
  <div
    className="video-info-box"
  >
    <Image
      as="img"
      circular={true}
      className="channel-image"
      src="http://via.placeholder.com/48x48"
      ui={true}
    />
    <div
      className="video-info"
    >
      <div
        className="channel-name"
      >
        Channel Name
      </div>
      <div
        className="video-publication-date"
      >
        Sat, Jul 18 2020
      </div>
    </div>
    <Button
      as="button"
      color="youtube"
    >
      10K Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <div>
          <p>
            Para 1
          </p>
          <p>
            Para 2
          </p>
          <p>
            Para 3
          </p>
          <p>
            Para 4
          </p>
          <p>
            Para 5
          </p>
        </div>
      </div>
      <Button
        as="button"
        compact={true}
        onClick={[Function]}
      >
        Show Less
      </Button>
    </div>
  </div>
  <Divider />
</Fragment>
`;
